jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/ConditionalHandler.java

```diff
@@ -0,0 +1,811 @@
+        }
+
+        @Override
+        public boolean test(Request request)
+        {
+            return _method.equals(request.getMethod());
+        }
+
+        @Override
+        public int hashCode()
+        {
+            return _method.hashCode();
+        }
+
+        @Override
+        public boolean equals(Object obj)
+        {
+            return obj instanceof MethodPredicate other && _method.equals(other._method);
+        }
+
+        @Override
+        public String toString()
+        {
+            return String.format("%s@%x{%s}", getClass().getSimpleName(), hashCode(), _method);
+        }
+    }
+
+    /**
+     * A {@link Predicate} over {@link Request} that tests a {@link PathSpec} against
+     * the {@link Request#getPathInContext(Request) pathInContext}.
+     * Using predicates in less efficient than using {@link ConditionalHandler#include(PathSpec...)}
+     * and {@link ConditionalHandler#exclude(PathSpec...)}, so this predicate should only be used
+     * if necessary to combine with other predicates.
+     *
+     */
+    public static class PathSpecPredicate implements Predicate<Request>
+    {
+        private final PathSpec _pathSpec;
+
+        public PathSpecPredicate(PathSpec pathSpec)
+        {
+            _pathSpec = Objects.requireNonNull(pathSpec);
+        }
+
+        @Override
+        public boolean test(Request request)
+        {
+            return _pathSpec.matches(Request.getPathInContext(request));
+        }
+
+        @Override
+        public int hashCode()
+        {
+            return _pathSpec.hashCode();
+        }
+
+        @Override
+        public boolean equals(Object obj)
+        {
+            return obj instanceof PathSpecPredicate other && _pathSpec.equals(other._pathSpec);
+        }
+
+        @Override
+        public String toString()
+        {
+            return String.format("%s@%x{%s}", getClass().getSimpleName(), hashCode(), _pathSpec);
+        }
+    }
+
+    /**
+     * An Abstract {@link ConditionalHandler}.  Implementations must provide
+     * both {@link #doHandle(Request, Response, Callback)} and
+     * {@link #doNotHandle(Request, Response, Callback)} implementations.
+     */
+    public abstract static class Abstract extends ConditionalHandler
+    {
+        protected Abstract()
+        {
+        }
+
+        protected Abstract(Handler nextHandler)
+        {
+            super(nextHandler);
+        }
+
+        protected Abstract(boolean dynamic, Handler nextHandler)
+        {
+            super(dynamic, nextHandler);
+        }
+    }
+
+    /**
+     * An implementation of {@link ConditionalHandler} that, if conditions are met, will not do any further
+     * handling by returning {@code false} from {@link #doHandle(Request, Response, Callback)}.
+     * Otherwise, the {@link #nextHandle(Request, Response, Callback) next handler} will be invoked.
+     */
+    public static class DontHandle extends ConditionalHandler
```

Comment:
Are not these classes the other way around?

If I want to create a Jetty XML file where I want to wrap my handler with a conditional so that my handler only gets called for GET to `*.dbimage`, I'd like to do (in code to avoid writing XML);

```java
Conditional cond = new ConditionalHandler.SkipNext();
cond.includeMethod("GET");
cond.includePath("*.dbimage");
cond.setHandler(new DBImageHandler());
```

Now, when the condition matches, it skips it.

Reply:
@sbordet These class are ConditionalHandlers.   If I make GzipHandler conditional, then when the conditions are met it gzips.    So in the same way, when the conditions are met Reject rejects, DontHandle doesn't handle and SkipNext skips the next one.
So in your example, the conditons apply to the action "skipNext".   It is the  action "skipNext" that applies (or not) to DBImageHandler.